pkg/controller/plan/adapter/ovirt/validator.go

```diff
@@ -106,7 +106,12 @@ func (r *Validator) StorageMapped(vmRef ref.Ref) (ok bool, err error) {
 		return
 	}
 	for _, da := range vm.DiskAttachments {
-		if !r.plan.Referenced.Map.Storage.Status.Refs.Find(ref.Ref{ID: da.Disk.StorageDomain}) {
+		if da.Disk.StorageType != "lun" {
+			if !r.plan.Referenced.Map.Storage.Status.Refs.Find(ref.Ref{ID: da.Disk.StorageDomain}) {
+				return
+			}
+		} else if len(da.Disk.Lun.LogicalUnits.LogicalUnit) > 0 && da.Disk.Lun.LogicalUnits.LogicalUnit[0].Address == "" {
+			// have LUN but without the relevant data
```

Comment:
yeah, it makes sense to do that but the question is why here - can't we filter out this information before saving it to the inventory then?

Reply:
filtering before saving - so the vm will be without the disk?

Reply:
oh that's even better - then we can produce a validation error!

Reply:
how so? empty map is allowed...

Reply:
right but the storage map is set later on - I'm referring to an earlier phase where we examine the vm from the source, if we see a diskless vm (e.g., a vm with a lun disk that we cannot migrate) then we can produce a validation error (not necessarily an 'error', maybe a 'warning in the general case and an error when the vm is set with a bootable disk that we cannot migrate)

Reply:
I still think it would be better to filter out the disk so we won't see invalid disks in the storage mapping and ideally also have a validation warning about this